### PR TITLE
upgrade(datadog-agent): Enable Remote Configuration by default

### PR DESCRIPTION
**Pull request by BaptisteFoy**
#### What this PR does / why we need it:
Enables Remote Config by default. Follows https://github.com/DataDog/datadog-agent/pull/17991

This inverts the previous logic: Remote Config is on by default, unless at least one of the values sets it to false, in which case Remote Config will be disabled

#### Checklist
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Review by arbll — APPROVED** [review]

File: charts/datadog/templates/_helpers.tpl [hunk 1]
```diff
@@ -814,20 +814,20 @@ false
 Returns whether Remote Configuration should be enabled in the agent
 */}}
 {{- define "datadog-remoteConfiguration-enabled" -}}
-{{- if or .Values.remoteConfiguration.enabled .Values.datadog.remoteConfiguration.enabled -}}
-true
-{{- else -}}
+{{- if or (not .Values.remoteConfiguration.enabled) (not .Values.datadog.remoteConfiguration.enabled) -}}
```
Comment (celenechang) [inline charts/datadog/templates/_helpers.tpl:817]: Could we make this more explicit by doing this instead?

```
{{- if and .Values.remoteConfiguration.enabled .Values.datadog.remoteConfiguration.enabled -}}
true
```

Also we have a question about intent - right now the logic is such that if either of the parameters is F, then the result will be F. 

F F -> F
F T -> F
T T -> T

However, in the README it says 

> `remoteConfiguration.enabled... Can be overriden by `datadog.remoteConfiguration.enabled` and `clusterAgent.admissionController.remoteInstrumentation.enabled`

Does this imply that F T should result in T?

Comment (BaptisteFoy) [reply charts/datadog/templates/_helpers.tpl:817]: As Remote Config is now on by default in the agent, we want to provide a parameter that lets us **disable** Remote Config is they want to; and we don't want to change two parameters to disable Remote Config.

This is why the logic has changed to give what you have (False having the priority) - we can't really remove parameters, sadly.


As for the simplification, yep this is better let me fix that

File: charts/datadog/README.md [hunk 2]
```diff
@@ -760,7 +760,7 @@ helm install <RELEASE_NAME> \
 | providers.gke.autopilot | bool | `false` | Enables Datadog Agent deployment on GKE Autopilot |
 | providers.gke.cos | bool | `false` | Enables Datadog Agent deployment on GKE with Container-Optimized OS (COS) |
 | registry | string | `"gcr.io/datadoghq"` | Registry to use for all Agent images (default gcr.io) |
-| remoteConfiguration.enabled | bool | `false` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. This will override `datadog.remoteConfiguration.enabled` and `clusterAgent.admissionController.remoteInstrumentation.enabled`. Preferred way to enable Remote Configuration. |
+| remoteConfiguration.enabled | bool | `true` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. Can be overriden by `datadog.remoteConfiguration.enabled` and `clusterAgent.admissionController.remoteInstrumentation.enabled`. Preferred way to enable Remote Configuration. |
```
Comment (celenechang) [inline charts/datadog/README.md:763]: Thanks for the explanation in the other comment! Could we make a small tweak here?

```suggestion
| remoteConfiguration.enabled | bool | `true` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. Can be overriden if `datadog.remoteConfiguration.enabled` or `clusterAgent.admissionController.remoteInstrumentation.enabled` is set to `false`. Preferred way to enable Remote Configuration. |
```

File: charts/datadog/README.md [hunk 3]
```diff
@@ -760,7 +760,7 @@ helm install <RELEASE_NAME> \
 | providers.gke.autopilot | bool | `false` | Enables Datadog Agent deployment on GKE Autopilot |
 | providers.gke.cos | bool | `false` | Enables Datadog Agent deployment on GKE with Container-Optimized OS (COS) |
 | registry | string | `"gcr.io/datadoghq"` | Registry to use for all Agent images (default gcr.io) |
-| remoteConfiguration.enabled | bool | `false` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. This will override `datadog.remoteConfiguration.enabled` and `clusterAgent.admissionController.remoteInstrumentation.enabled`. Preferred way to enable Remote Configuration. |
+| remoteConfiguration.enabled | bool | `true` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. Can be overriden if `datadog.remoteConfiguration.enabled` or `clusterAgent.admissionController.remoteInstrumentation.enabled` is set to `false`. Preferred way to enable Remote Configuration. Preferred way to enable Remote Configuration. |
```
Comment (celenechang) [inline charts/datadog/README.md:763]:
```suggestion
| remoteConfiguration.enabled | bool | `true` | Set to true to enable remote configuration on the Cluster Agent (if set) and the node agent. Can be overriden if `datadog.remoteConfiguration.enabled` or `clusterAgent.admissionController.remoteInstrumentation.enabled` is set to `false`. Preferred way to enable Remote Configuration. |
```

**Review by celenechang — APPROVED** [review]

**Pull request merged**